cmd/restic/main.go

```diff
@@ -139,15 +139,7 @@ func main() {
 	case errors.IsFatal(err):
 		fmt.Fprintf(os.Stderr, "%v\n", err)
 	case err != nil:
-		fmt.Fprintf(os.Stderr, "%+v\n", err)
-
-		if logBuffer.Len() > 0 {
-			fmt.Fprintf(os.Stderr, "also, the following messages were logged by a library:\n")
-			sc := bufio.NewScanner(logBuffer)
-			for sc.Scan() {
-				fmt.Fprintln(os.Stderr, sc.Text())
-			}
-		}
+		fmt.Fprintf(os.Stderr, "Fatal: %v\n", err)
```

Comment:
This was a big swing by me - happy to go a different way.

Previously, `IsFatal` was used to flag "expected errors that will bubble up to main.go" - and thus any other errors should get all this debugging info added, since they were unexpected.

But if we are moving away from fatal errors and just using the original errors instead, we can either (a) add a case statement for each expected error, or (b) drop this extra debugging step for unexpected errors. I chose the latter, but am happy to do the former (or some other idea).

Reply:
That will definitely help moving forwards. But in my opinion it is too invasive to include in a patch version. So we'll have to change this separately from adding the exit code if the latter should be included in 0.17.1

Reply:
(I myself do not need this to land on a specific schedule - so if you want to save some of the changes for 0.18, I won't be annoyed. :smile:)

Let's recap the options here:
- Don't wrap as a FatalError. Just wrap with `fmt.Errorf("%w")`
  - This causes the full stacktrace to be printed to the console for this error, since main.go no longer sees it as an expected error.
  - So option (1): drop stacktraces from main.go for all errors (what I've done here)
  - Or option (2): drop stacktraces from main.go for just  `repository.ErrNoKeyFound` errors
- Or option (3): Allow FatalErrors to properly wrap and hold an inside error. (my original version)

It sounds like option 3 you don't like in general. Option 1 you don't love in a patch version.

But option 2 is maybe fine as a short term fix? And maybe in 0.18 and/or as we grow more "expected error" cases, we can revisit.

Reply:
OK, I added option 2 (just a manual check for this one error) to this PR.